http/src/test/java/com/hazelcast/jet/contrib/http/HttpListenerSinkTest.java

```diff
@@ -578,7 +556,11 @@ void postMessages(IQueue<String> sourceQueue, int messageCount) {
         for (int i = 0; i < messageCount; i++) {
             sourceQueue.offer(String.format(pattern, i));
         }
-        assertEqualsEventually(sourceQueue::size, 0);
+    }
+
+    void assertReceivedCountEventually(int expected) {
+        assertTrueEventually(() -> assertEquals(expected,
+                job.getMetrics().get(RECEIVED_COUNT).stream().mapToLong(Measurement::value).sum()));
```

Comment:
I think it would be better if we filter out the metrics for the edges whose destination vertex is not the sink.